erpc_c/port/erpc_sysgpio.c

```diff
@@ -44,8 +46,8 @@ int gpio_export(int gpio)
         }
         else
         {
-            ret = sprintf(sysgpio, "%d", gpio);
-            if (ret >= 0)
+            ret = snprintf(sysgpio, SYSGPIO_LEN, "%d", gpio);
+            if ((ret >= 0) && (ret < SYSGPIO_LEN))
             {
                 if (strlen(sysgpio) != write(fd, sysgpio, strlen(sysgpio)))
```

Comment:
I would replace both strlen(sysgpio) with ret

Reply:
Very nice catch. Thank you ;)